### ISSUE TITLE
bug: switchNetwork wrong trigger

### ISSUE THREAD

**Issue by BeycanDeveloper** [wevm/viem#1550]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.19.9

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://codesandbox.io/p/devbox/elegant-ardinghelli-n53nwj?file=%2Fsrc%2Findex.mjs%3A18%2C10

### Anything else?

If you use the network switching feature during the connection process and reject the network switching instead of accepting it, everything will work fine at first.

However, when you try the connection for the second time, the network switching screen appears, but w3m does not wait for the response from the wallet and terminates the process by giving the warning in the picture below.

![image](https://github.com/wevm/viem/assets/50718965/78668867-05ef-4462-9542-175d51a4371f)

When you press connect for the first time and reject it on the change screen, it will be written to the console as "Error". When you press connect for the second time, before you do anything, a warning will appear as shown in the picture below. Therefore, from now on, no action will be taken by the user whether you approve or reject it. Because it has already been received as rejection.

![image](https://github.com/wevm/viem/assets/50718965/99f6840a-2672-4924-b39e-ad00e998bae0)

List of related npm package versions
"@wagmi/core": "^1.4.7"
"@web3modal/wagmi": "^3.3.2"
"viem": "^1.19.9"

**Comment by jxom** [comment]
Thanks for reporting. Can you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) via https://new.viem.sh or a separate repo? It will save us a lot of time reproducing your config & environment, and trying to reproduce your issue. See [Why reproductions are required](https://antfu.me/posts/why-reproductions-are-required).

Please reopen this issue when a reproduction is added.

**Comment by imaksp** [comment]
Hi, it seems link is already provided & I can reproduce using provided [codesandbox](https://codesandbox.io/p/devbox/elegant-ardinghelli-n53nwj?file=%2Fsrc%2Findex.mjs%3A18%2C10) link.

but it seems issue may not be specific to this lib, as we are using it in Angular with only wagmi/core & Viem (without wrapper lib like @web3modal/wagmi ) & we did not face such issue, also with only wagmi/core & viem you don't need to call switchNetwork/chain manually , you can pass intended chain in [connect function](https://wagmi.sh/core/actions/connect#chainid-optional) & it will automatically handle this. 

& This is just a quick guess.

**Comment by BeycanDeveloper** [comment]
@imaksp w3m has a parameter called defaultChain but it doesn't change the chain.

**Comment by jxom** [comment]
Might be worth raising an issue on the Web3Modal GitHub as this is not an issue related to the Viem core.

**Comment by BeycanDeveloper** [comment]
The error I mentioned is thrown from here: at EIP6963Connector.switchChain

![image](https://github.com/wevm/viem/assets/50718965/12764257-7fee-4916-89d0-942a707d8e28)

@jxom  When I opened this issue before you, they said they were just throwing out the mistakes they received from you. Isn't it clearly visible in the picture?

**Comment by BeycanDeveloper** [comment]
I got it because when I use it directly on metamask everything works fine. Now I cloned viem and will try it on it. If the problem is here, I commit and create a pull request.

Because to be honest, I don't know which one I have since there are so many different packages until I get to the end. But the only thing I'm sure of is that it's not directly in metamask. Because its API works properly.
